src/toil/common.py

```diff
@@ -498,20 +568,116 @@ def parser_with_common_options(
     return parser
 
 
-def addOptions(parser: ArgumentParser, config: Optional[Config] = None, jobstore_as_flag: bool = False) -> None:
+def make_coordinationdir_check_action() -> Type[_StoreAction]:
+    class CheckCoordinationDir(_StoreAction):
+        def __call__(self, parser: Any, namespace: Any, values: Any, option_string: Any=None) -> None:
+            coordination_dir = values
+            if coordination_dir is not None:
+                coordination_dir = os.path.abspath(coordination_dir)
+                if not os.path.exists(coordination_dir):
+                    raise RuntimeError(
+                        f"The path provided to --coordinationDir ({coordination_dir}) does not exist.")
+            setattr(namespace, self.dest, values)
+    return CheckCoordinationDir
```

Comment:
Can't this just be a normal class? Why do we need the factory function if it takes no arguments?